.travis.yml

```diff
@@ -6,4 +6,44 @@ cache: yarn
 
 script:
   - bin/check_required_files_present
-  - yarn test
+  - |
+    invalid_old=""
+    invalid_new=""
+    forgotten=""
+
+    for json in exercises/*/canonical-data.json; do
+      dir=$(dirname $json)
+      ex=$(basename $dir)
+
+      if [ -f "$dir/USE_OLD_SCHEMA" ]; then
+        if ! ajv -s old-schema.json -d $json; then
+          invalid_old="$invalid_old $ex"
+        fi
+        # If we have converted to new,
+        # make sure we don't forget to remove USE_OLD_SCHEMA file!
+        if ajv -s canonical-schema.json -d $json 2>/dev/null; then
+          forgotten="$forgotten $ex"
+        fi
+      else
+        if ! ajv -s canonical-schema.json -d $json; then
```

Comment:
this could have been an elsif but I didn't think it would make the code sufficiently clear.